src/observer/utils.py

```diff
@@ -24,6 +24,12 @@ def nth(x, n):
 first = partial(nth, n=0)
 second = partial(nth, n=1)
 third = partial(nth, n=2)
+last = partial(nth, n=-1)
+
+def deepcopy(d):
+    # copy.deepcopy is exceptionally slow!
+    # TODO: replace this wrapper with a faster implementation
+    return copy.deepcopy(d)
```

Comment:
🐢

Reply:
that's an interesting trick. 

Can you elaborate on your testing methodology and results? Were you testing the speed of this project or `deepcopy` in particular?

This PR cuts the runtime of a full regeneration in half and it's given me the optimisation bug. I was daydreaming this morning over my breakfast about how I could speed up `et3` with lisp and macros ...

Reply:
Sure. I was basing it on a simple local test. I had a fairly large `json` file (around 37,000+ lines) with various nested keys values to exercise `deepcopy`s recursion a bit, parsed it, then feed the `dict` to two functions, one was using `deepcopy` one was using `pickle`.

The profiler consistently returned values with similar gains:

`with_deepcopy()`: 102ms
`with_pickle()`: 9.2ms

I have used it in the past, though back then you had to use `cPickle`, but thats used under the hood with Python 3 now anyway.

Reply:
Yer looks like those results are Python 2 based though. I would be interested to see how much it changed the speed of `observer`s regeneration time, if any.